cloud/aws/ses/mode/emails.pm

```diff
@@ -0,0 +1,240 @@
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'metrics', type => 3, cb_prefix_output => 'prefix_metric_output', cb_long_output => 'long_output',
+          message_multiple => 'All emails metrics are ok', indent_long_output => '    ',
+            group => [
+                { name => 'statistics', display_long => 1, cb_prefix_output => 'prefix_statistics_output',
+                  message_multiple => 'All metrics are ok', type => 1, skipped_code => { -10 => 1 } },
+            ]
+        }
+    ];
+
+    foreach my $metric (keys %metrics_mapping) {
+        my $entry = {
+            label => $metrics_mapping{$metric}->{label},
+            set => {
+                key_values => [ { name => $metric }, { name => 'timeframe' }, { name => 'display' } ],
+                closure_custom_calc => $self->can('custom_metric_calc'),
+                closure_custom_calc_extra_options => { metric => $metric },
+                closure_custom_output => $self->can('custom_metric_output'),
+                closure_custom_perfdata => $self->can('custom_metric_perfdata'),
+                closure_custom_threshold_check => $self->can('custom_metric_threshold'),
+            }
+        };
+        push @{$self->{maps_counters}->{statistics}}, $entry;
+    }
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+        'name:s@' => { name => 'name' },
+    });
+
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+
+    if (!defined($self->{option_results}->{name}) || $self->{option_results}->{name} eq '') {
+        $self->{output}->add_option_msg(short_msg => "Need to specify --name option.");
+        $self->{output}->option_exit();
+    }
+
+    foreach my $instance (@{$self->{option_results}->{name}}) {
+        if ($instance ne '') {
+            push @{$self->{aws_instance}}, $instance;
+        }
+    }
+
+    $self->{aws_timeframe} = defined($self->{option_results}->{timeframe}) ? $self->{option_results}->{timeframe} : 172800;
+    $self->{aws_period} = defined($self->{option_results}->{period}) ? $self->{option_results}->{period} : 86400;
+    $self->{aws_statistics} = ['Average'];
+
+    foreach my $metric (keys %metrics_mapping) {
+        next if (defined($self->{option_results}->{filter_metric}) && $self->{option_results}->{filter_metric} ne ''
+            && $metric !~ /$self->{option_results}->{filter_metric}/);
+        push @{$self->{aws_metrics}}, $metric;
+    }
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my %metric_results;
+    foreach my $instance (@{$self->{aws_instance}}) {
+        $metric_results{$instance} = $options{custom}->cloudwatch_get_metrics(
+            namespace => 'AWS/SES',
+            dimensions => [],
```

Comment:
Probably not good see https://github.com/centreon/centreon-plugins/blob/15111163ac65e4fea24c04a65be388fe4a6f9977/cloud/aws/sqs/mode/queues.pm#L174 for an example when using a name filter

Reply:
I left that field empty because that is what I saw when I listed the metrics for SES. Example:
```
aws cloudwatch list-metrics --namespace="AWS/SQS"                                                                                                   {
    "Metrics": [
        {
            "Namespace": "AWS/SQS",
            "Dimensions": [
                {
                    "Name": "QueueName",
                    "Value": "ProdSFAction"
                }
            ],
...
```
But with SES:
```
aws cloudwatch list-metrics --namespace="AWS/SES"
{
    "Metrics": [
        {
            "Namespace": "AWS/SES",
            "Dimensions": [],
            "MetricName": "Send"
        },
...
```

Reply:
https://docs.aws.amazon.com/ses/latest/DeveloperGuide/event-publishing-retrieving-cloudwatch.html#event-publishing-retrieving-cloudwatch-dimensions

Should I add a 'Dimension' option ?

Reply:
Ok, let it like that. But try to add the capability to set dimension (see https://github.com/centreon/centreon-plugins/blob/71e3a9fe7ea6e61242499eac1cbbc8a134d477a7/cloud/aws/cloudwatch/mode/getmetrics.pm#L96)